### PR TITLE
Be sure to not reference to state when selected rows are modified

### PR DESCRIPTION
**Pull request by acollin1**

### CODE REVIEW

**Comment by acollin1** [comment]
Demo: https://coveo.github.io/react-vapor/fix-reference-error-in-facetconnected-state/

File: src/components/facets/FacetConnected.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ const mapStateToProps = (state: IReactVaporState, ownProps: IFacetOwnProps): IFa
 
 const getSelectedRows = (rows: IFacet[], selected: IFacet[]): IFacet[] => _.without(_.map(rows, (row: IFacet) => {
     const element = _.findWhere(selected, {name: row.name});
-    return element ? _.extend(element, row) : null;
+    return element ? _.extend({}, element, row) : null;
```
Comment (gdostie) [inline src/components/facets/FacetConnected.tsx]: I think `return element && {...element, ...row};` would do the same and is shorter.

File: src/components/facets/FacetReducers.ts [hunk 2]
```diff
@@ -46,24 +46,32 @@ const changeFacet = (state: IFacetState, action: (IReduxAction<IReduxActionsPayl
         return state;
     }
 
-    let selected = state.selected;
+    let selected = _.clone(state.selected);
     if (_.some(state.selected, (facetRow: IFacet) => facetRow.name === action.payload.facetRow.name)) {
         const selectedIndex: number = _.findIndex(state.selected, {name: action.payload.facetRow.name});
         if (!selected[selectedIndex].exclude && action.payload.facetRow.exclude) {
-            selected[selectedIndex] = {...selected[selectedIndex], exclude: true};
```
Comment (GermainBergeron) [inline src/components/facets/FacetReducers.ts]: This won't modify the original object

File: src/components/facets/FacetReducers.ts [hunk 3]
```diff
@@ -46,24 +46,32 @@ const changeFacet = (state: IFacetState, action: (IReduxAction<IReduxActionsPayl
...
+                name: selected[selectedIndex].name,
+                formattedName: selected[selectedIndex].formattedName,
+                exclude: true,
+            };
         } else {
             selected = _.reject(state.selected, (facetRow: IFacet, index: number) => index === selectedIndex);
         }
     } else {
         selected = [
-            action.payload.facetRow,
```
Comment (GermainBergeron) [inline src/components/facets/FacetReducers.ts]:
```
selected = [
 {...action.payload.facetRow},
...state.selected,
];
```

Comment (acollin1) [reply src/components/facets/FacetReducers.ts]: It is to not pass the count attr. Maybe I can just do an ..._.omit(action.payload.facetRow, 'count')

Comment (GermainBergeron) [reply src/components/facets/FacetReducers.ts]: Then I prefer your solution

**Pull request closed without merging**